pkg/config/config.go

```diff
@@ -837,6 +853,45 @@ func (c *config) Validate() error {
 			c.Auth.External[ext] = provider
 		}
 	}
+	// Validate Third-Party Auth config
+	if c.ThirdPartyAuth.Enabled {
+		allDisabled := true
+		allDisabled = allDisabled && c.ThirdPartyAuth.FirebaseProjectID == ""
+		allDisabled = allDisabled && c.ThirdPartyAuth.Auth0Tenant == ""
+		allDisabled = allDisabled && c.ThirdPartyAuth.Auth0TenantRegion == ""
+		allDisabled = allDisabled && c.ThirdPartyAuth.AWSCognitoUserPoolID == ""
+		allDisabled = allDisabled && c.ThirdPartyAuth.AWSCognitoUserPoolRegion == ""
```

Comment:
This is unfortunately verbose... I really want to start using https://github.com/go-playground/validator for our config validation soon.

Reply:
Tried to simplify it by having separate validate functions on the structs.